controllers/machine_controller.go

```diff
@@ -310,3 +390,25 @@ func (r *MachineReconciler) reconcileDeleteExternal(ctx context.Context, m *clus
 func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"MachineSet", "Cluster"})
 }
+
+// Writer implements io.Writer interface as a pass-through for klog.
+type Writer struct {
+	logFunc func(args ...interface{})
+}
+
+// Write passes string(p) into Writer's logFunc and always returns len(p)
+func (w Writer) Write(p []byte) (n int, err error) {
+	w.logFunc(string(p))
+	return len(p), nil
+}
+
+// onPodDeletedOrEvicted is called by drain.Helper, when the pod has been deleted or evicted
+func onPodDeletedOrEvicted(pod *corev1.Pod, usingEviction bool) {
+	var verbStr string
+	if usingEviction {
+		verbStr = "evicted"
+	} else {
+		verbStr = "deleted"
+	}
+	klog.Infof("pod %s/%s %s\n", pod.Namespace, pod.Name, verbStr)
```

Comment:
This is only visible in the pod's logs. Is that useful?

Reply:
I would consider this information mostly debug-level.  I'm not really sure what the UX expectation is here, but I think looking at the pod's logs is reasonable for most things.